raimitigations/automitigator/evaluator.py

```diff
@@ -0,0 +1,208 @@
+
+        :param sklearn.pipeline.Pipeline pipeline: The pipeline to add the step to
+        :param step: The step to add
+        """
+        if self.pipeline is None:
+            self.pipeline = Pipeline([step])
+        else:
+            self.pipeline.steps.append(step)
+
+    def evaluate(self, train_x, train_y, search_config):
+        """
+        Evaluates a given set of mitigations on a given dataset.
+
+        :param train_x: The training data
+        :param train_y: The training labels
+        :param dict search_config: The search configuration
+
+        :return: The results of the evaluation
+        :rtype: dict
+        """
+        # Sample config
+        # config: {'search_space':
+        #           {'cohort': 'all',
+        #            'mitigations':
+        #               {'action0': {'type': 0, 'strategy': 0, 'name': 'rebalancer'}}}}
+
+        self.pipeline = None
+        search_space = search_config["search_space"]
+        cohort = search_space["cohort"]
+        if cohort == "all":
+            return self.mitigate_full_dataset(train_x, train_y, search_space)
+        else:
+            raise ValueError(f"Unknown cohort type {cohort}")
+
+    def _process_feature_selector(self, selector_type):
+        """
+        Process the feature selector
+
+        :param dict selector_type: The feature selector configuration
+
+        :raises ValueError: If the feature selector is unknown
+        """
+        selector_name = selector_type["selector_name"]
+        if selector_name == "sequential_selector":
+            self._pipeline_append(("sequential_selector", dp.SeqFeatSelection()))
+        elif selector_name == "correlated_feature_selector":
+            self._pipeline_append(
+                (
+                    "correlated_feature_selector",
+                    dp.CorrelatedFeatures(
+                        num_corr_th=selector_type["num_corr_th"],
+                        num_pvalue_th=selector_type["num_pvalue_th"],
+                        cat_corr_th=selector_type["cat_corr_th"],
+                        cat_pvalue_th=selector_type["cat_pvalue_th"],
+                    ),
+                )
+            )
+        elif selector_name == "catboost_selector":
+            self._pipeline_append(
+                (
+                    "catboost_selector",
+                    dp.CatBoostSelection(
+                        test_size=selector_type["test_size"],
+                        algorithm=selector_type["algorithm"],
+                        steps=selector_type["steps"],
+                        verbose=False,
+                    ),
+                )
+            )
+
+    def mitigate_full_dataset(self, train_x, train_y, search_space):
+        """
+        Evaluates a given set of mitigations on a given dataset.
+
+        :param train_x: The training data
+        :param train_y: The training labels
+        :param dict search_space: The search configuration
```

Comment:
This ``search_space`` parameter needs a more in-depth documentation. I more detailed docstring for it, or some examples of different use cases in a notebook inside the ``notebooks/`` folder would help a lot.

Reply:
Agreed. A separate notebook with usage information is planned for separately.